src/boundary_conditions/src/pressure_pinning.C

```diff
@@ -66,16 +69,57 @@ namespace GRINS
     return;
   }
 
-  void PressurePinning::pin_value( libMesh::DiffContext &context, 
+  void PressurePinning::check_pin_location( const libMesh::MeshBase& mesh )
+  {
```

Comment:
You need to reset _pinned_elem_id to invalid_id here, no?  Otherwise this class will work the first time it's used but may fail after AMR.